cpp/include/raft/detail/layout_padded_general.hpp

```diff
@@ -122,17 +119,44 @@ struct layout_padded_general {
 
     MDSPAN_INLINE_FUNCTION
     constexpr mapping(Extents const& e) noexcept
-      : layout_stride::mapping<Extents>{
-          e,
-          StorageOrder == StorageOrderType::row_major_t
-            ? padded_row_major_strides(ByteAlignment, sizeof(ValueType), e)
-            : padded_col_major_strides(ByteAlignment, sizeof(ValueType), e)}
+      : layout_stride::mapping<Extents>{e,
+                                        StorageOrder == StorageOrderType::row_major_t
+                                          ? padded_row_major_strides(Alignment, e)
+                                          : padded_col_major_strides(Alignment, e)}
     {
     }
 
     //--------------------------------------------------------------------------------
   };
 };
 
+template <class ElementType, size_t Alignment>
+struct aligned_accessor {
+  using offset_policy               = aligned_accessor;
+  using element_type                = ElementType;
+  using reference                   = ElementType&;
+  using pointer                     = ElementType*;
+  static constexpr size_t alignment = Alignment;
+
+  constexpr aligned_accessor() noexcept = default;
+
+  MDSPAN_TEMPLATE_REQUIRES(
+    class OtherElementType,
+    size_t OtherAlignment,
+    /* requires */
+    (_MDSPAN_TRAIT(is_convertible,
+                   typename aligned_accessor<OtherElementType, OtherAlignment>::element_type (*)[],
+                   element_type (*)[]) &&
+     alignment == OtherAlignment))
+  MDSPAN_INLINE_FUNCTION
+  constexpr aligned_accessor(aligned_accessor<OtherElementType, OtherAlignment>) noexcept {}
+
+  MDSPAN_INLINE_FUNCTION
+  constexpr pointer offset(pointer p, size_t i) const noexcept { return p + i; }
+
+  MDSPAN_FORCE_INLINE_FUNCTION
+  constexpr reference access(pointer p, size_t i) const noexcept { return p[i]; }
```

Comment:
[This is not strictly required for correctness, but I'm guessing you won't get any benefit out of `aligned_accessor` unless you do something here.]

This would be the point at which I make assertions about alignment.  For example:
```c++
constexpr reference access(data_handle_type p, size_t i) const noexcept {
#if defined(__cpp_lib_assume_aligned)
    return (std::assume_aligned<alignment>(p))[i];
#elif defined(__GNUC__)
    return (__builtin_assume_aligned(p, alignment))[i];
#else
    return p[i];
#endif
}
```

Reply:
Yes, for now I was just switching to the right accessor class in order to allow for both static and runtime checks later. Thanks for example, I'll work that in.

Reply:
@mfoerste4 Some compilers let one express alignment by annotating the pointer; e.g., `double* __attribute__(align_value( alignment ))` -- I haven't experimented with that, but I personally would prefer using the C++ Standard way (`std::assume_aligned`, as above) to the list of `ifdef`s.